Build failure: gdtoolkit

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

see hydra failure

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/305752951

### Relevant log output

```console
=================================== FAILURES ===================================
__________________________ test_valid_file_formatting __________________________

tmp_path = PosixPath('/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test_valid_file_formatting0')

    def test_valid_file_formatting(tmp_path):
        dummy_file = write_file(tmp_path, "script.gd", "pass")
        outcome = subprocess.run(["gdformat", dummy_file], check=False, capture_output=True)
        assert outcome.returncode == 0, outcome.stderr.decode()
        assert len(outcome.stdout.decode().splitlines()) == 2
>       assert len(outcome.stderr.decode().splitlines()) == 0
E       AssertionError: assert 2 == 0
E        +  where 2 = len(['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'])
E        +    where ['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'] = <built-in method splitlines of str object at 0x110819110>()
E        +      where <built-in method splitlines of str object at 0x110819110> = '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.splitlines
E        +        where '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = <built-in method decode of bytes object at 0x110818f60>()
E        +          where <built-in method decode of bytes object at 0x110818f60> = b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.decode
E        +            where b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = CompletedProcess(args=['gdformat', '/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test_vali...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n').stderr

tests/formatter/test_executable.py:11: AssertionError
_________________________ test_valid_files_formatting __________________________

tmp_path = PosixPath('/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test_valid_files_formatting0')

    def test_valid_files_formatting(tmp_path):
        dummy_file = write_file(tmp_path, "script.gd", "pass")
        dummy_file_2 = write_file(tmp_path, "script2.gd", "pass;pass")
        outcome = subprocess.run(
            ["gdformat", dummy_file, dummy_file_2], check=False, capture_output=True
        )
        assert outcome.returncode == 0
        assert len(outcome.stdout.decode().splitlines()) == 3
>       assert len(outcome.stderr.decode().splitlines()) == 0
E       AssertionError: assert 2 == 0
E        +  where 2 = len(['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'])
E        +    where ['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'] = <built-in method splitlines of str object at 0x110819470>()
E        +      where <built-in method splitlines of str object at 0x110819470> = '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.splitlines
E        +        where '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = <built-in method decode of bytes object at 0x1108192c0>()
E        +          where <built-in method decode of bytes object at 0x1108192c0> = b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.decode
E        +            where b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = CompletedProcess(args=['gdformat', '/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test_vali...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n').stderr

tests/formatter/test_executable.py:22: AssertionError
______________________ test_valid_formatted_file_checking ______________________

tmp_path = PosixPath('/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test_valid_formatted_file_chec0')

    def test_valid_formatted_file_checking(tmp_path):
        dummy_file = write_file(tmp_path, "script.gd", "pass\n")
        outcome = subprocess.run(
            ["gdformat", "--check", dummy_file], check=False, capture_output=True
        )
        assert outcome.returncode == 0
        assert len(outcome.stdout.decode().splitlines()) == 1
>       assert len(outcome.stderr.decode().splitlines()) == 0
E       AssertionError: assert 2 == 0
E        +  where 2 = len(['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'])
E        +    where ['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'] = <built-in method splitlines of str object at 0x1108197d0>()
E        +      where <built-in method splitlines of str object at 0x1108197d0> = '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.splitlines
E        +        where '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = <built-in method decode of bytes object at 0x110819620>()
E        +          where <built-in method decode of bytes object at 0x110819620> = b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.decode
E        +            where b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = CompletedProcess(args=['gdformat', '--check', '/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n').stderr

tests/formatter/test_executable.py:61: AssertionError
_____________________ test_valid_unformatted_file_checking _____________________

tmp_path = PosixPath('/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test_valid_unformatted_file_ch0')

    def test_valid_unformatted_file_checking(tmp_path):
        dummy_file = write_file(tmp_path, "script.gd", "pass;var x")
        outcome = subprocess.run(
            ["gdformat", "--check", dummy_file], check=False, capture_output=True
        )
        assert outcome.returncode != 0
        assert len(outcome.stdout.decode().splitlines()) == 0
>       assert len(outcome.stderr.decode().splitlines()) == 2
E       AssertionError: assert 4 == 2
E        +  where 4 = len(['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...1/pytest-0/test_valid_unformatted_file_ch0/script.gd', '1 file would be reformatted, 0 files would be left unchanged.'])
E        +    where ['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...1/pytest-0/test_valid_unformatted_file_ch0/script.gd', '1 file would be reformatted, 0 files would be left unchanged.'] = <built-in method splitlines of str object at 0x115ec9b50>()
E        +      where <built-in method splitlines of str object at 0x115ec9b50> = '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...d1/pytest-0/test_valid_unformatted_file_ch0/script.gd\n1 file would be reformatted, 0 files would be left unchanged.\n'.splitlines
E        +        where '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...d1/pytest-0/test_valid_unformatted_file_ch0/script.gd\n1 file would be reformatted, 0 files would be left unchanged.\n' = <built-in method decode of bytes object at 0x115ec2e10>()
E        +          where <built-in method decode of bytes object at 0x115ec2e10> = b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...d1/pytest-0/test_valid_unformatted_file_ch0/script.gd\n1 file would be reformatted, 0 files would be left unchanged.\n'.decode
E        +            where b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...d1/pytest-0/test_valid_unformatted_file_ch0/script.gd\n1 file would be reformatted, 0 files would be left unchanged.\n' = CompletedProcess(args=['gdformat', '--check', '/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-...1/pytest-0/test_valid_unformatted_file_ch0/script.gd\n1 file would be reformatted, 0 files would be left unchanged.\n').stderr

tests/formatter/test_executable.py:71: AssertionError
_______________________ test_parsing_valid_file_succeeds _______________________

tmp_path = PosixPath('/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test_parsing_valid_file_succee0')

    def test_parsing_valid_file_succeeds(tmp_path):
        dummy_file = write_file(tmp_path, "script.gd", "pass")
        outcome = subprocess.run(["gdparse", dummy_file], check=False, capture_output=True)
        assert outcome.returncode == 0
        assert len(outcome.stdout.decode().splitlines()) == 0
>       assert len(outcome.stderr.decode().splitlines()) == 0
E       AssertionError: assert 2 == 0
E        +  where 2 = len(['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'])
E        +    where ['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'] = <built-in method splitlines of str object at 0x1108192c0>()
E        +      where <built-in method splitlines of str object at 0x1108192c0> = '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.splitlines
E        +        where '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = <built-in method decode of bytes object at 0x110819470>()
E        +          where <built-in method decode of bytes object at 0x110819470> = b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.decode
E        +            where b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = CompletedProcess(args=['gdparse', '/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test_parsi...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n').stderr

tests/parser/test_executable.py:11: AssertionError
______________________ test_parsing_valid_files_succeeds _______________________

tmp_path = PosixPath('/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test_parsing_valid_files_succe0')

    def test_parsing_valid_files_succeeds(tmp_path):
        dummy_file = write_file(tmp_path, "script.gd", "pass")
        dummy_file2 = write_file(tmp_path, "script2.gd", "pass;pass")
        outcome = subprocess.run(
            ["gdparse", dummy_file, dummy_file2], check=False, capture_output=True
        )
        assert outcome.returncode == 0
        assert len(outcome.stdout.decode().splitlines()) == 0
>       assert len(outcome.stderr.decode().splitlines()) == 0
E       AssertionError: assert 2 == 0
E        +  where 2 = len(['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'])
E        +    where ['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'] = <built-in method splitlines of str object at 0x110818f60>()
E        +      where <built-in method splitlines of str object at 0x110818f60> = '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.splitlines
E        +        where '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = <built-in method decode of bytes object at 0x110819110>()
E        +          where <built-in method decode of bytes object at 0x110819110> = b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.decode
E        +            where b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = CompletedProcess(args=['gdparse', '/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test_parsi...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n').stderr

tests/parser/test_executable.py:22: AssertionError
__________________ test_pretty_printing_valid_files_succeeds ___________________

tmp_path = PosixPath('/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test_pretty_printing_valid_fil0')

    def test_pretty_printing_valid_files_succeeds(tmp_path):
        dummy_file = write_file(tmp_path, "script.gd", "pass")
        dummy_file2 = write_file(tmp_path, "script2.gd", "pass;pass")
        outcome = subprocess.run(
            ["gdparse", "-p", dummy_file, dummy_file2], check=False, capture_output=True
        )
        assert outcome.returncode == 0
        assert len(outcome.stdout.decode().splitlines()) > 0
>       assert len(outcome.stderr.decode().splitlines()) == 0
E       AssertionError: assert 2 == 0
E        +  where 2 = len(['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'])
E        +    where ['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'] = <built-in method splitlines of str object at 0x110819980>()
E        +      where <built-in method splitlines of str object at 0x110819980> = '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.splitlines
E        +        where '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = <built-in method decode of bytes object at 0x110819b30>()
E        +          where <built-in method decode of bytes object at 0x110819b30> = b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.decode
E        +            where b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = CompletedProcess(args=['gdparse', '-p', '/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n').stderr

tests/parser/test_executable.py:33: AssertionError
=============================== warnings summary ===============================
gdtoolkit/parser/parser.py:8
  /private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/source/gdtoolkit/parser/parser.py:8: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.
    import pkg_resources

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/formatter/test_executable.py::test_valid_file_formatting - AssertionError: assert 2 == 0
 +  where 2 = len(['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'])
 +    where ['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'] = <built-in method splitlines of str object at 0x110819110>()
 +      where <built-in method splitlines of str object at 0x110819110> = '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.splitlines
 +        where '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = <built-in method decode of bytes object at 0x110818f60>()
 +          where <built-in method decode of bytes object at 0x110818f60> = b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.decode
 +            where b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = CompletedProcess(args=['gdformat', '/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test_vali...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n').stderr
FAILED tests/formatter/test_executable.py::test_valid_files_formatting - AssertionError: assert 2 == 0
 +  where 2 = len(['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'])
 +    where ['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'] = <built-in method splitlines of str object at 0x110819470>()
 +      where <built-in method splitlines of str object at 0x110819470> = '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.splitlines
 +        where '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = <built-in method decode of bytes object at 0x1108192c0>()
 +          where <built-in method decode of bytes object at 0x1108192c0> = b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.decode
 +            where b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = CompletedProcess(args=['gdformat', '/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test_vali...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n').stderr
FAILED tests/formatter/test_executable.py::test_valid_formatted_file_checking - AssertionError: assert 2 == 0
 +  where 2 = len(['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'])
 +    where ['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'] = <built-in method splitlines of str object at 0x1108197d0>()
 +      where <built-in method splitlines of str object at 0x1108197d0> = '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.splitlines
 +        where '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = <built-in method decode of bytes object at 0x110819620>()
 +          where <built-in method decode of bytes object at 0x110819620> = b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.decode
 +            where b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = CompletedProcess(args=['gdformat', '--check', '/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n').stderr
FAILED tests/formatter/test_executable.py::test_valid_unformatted_file_checking - AssertionError: assert 4 == 2
 +  where 4 = len(['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...1/pytest-0/test_valid_unformatted_file_ch0/script.gd', '1 file would be reformatted, 0 files would be left unchanged.'])
 +    where ['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...1/pytest-0/test_valid_unformatted_file_ch0/script.gd', '1 file would be reformatted, 0 files would be left unchanged.'] = <built-in method splitlines of str object at 0x115ec9b50>()
 +      where <built-in method splitlines of str object at 0x115ec9b50> = '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...d1/pytest-0/test_valid_unformatted_file_ch0/script.gd\n1 file would be reformatted, 0 files would be left unchanged.\n'.splitlines
 +        where '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...d1/pytest-0/test_valid_unformatted_file_ch0/script.gd\n1 file would be reformatted, 0 files would be left unchanged.\n' = <built-in method decode of bytes object at 0x115ec2e10>()
 +          where <built-in method decode of bytes object at 0x115ec2e10> = b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...d1/pytest-0/test_valid_unformatted_file_ch0/script.gd\n1 file would be reformatted, 0 files would be left unchanged.\n'.decode
 +            where b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...d1/pytest-0/test_valid_unformatted_file_ch0/script.gd\n1 file would be reformatted, 0 files would be left unchanged.\n' = CompletedProcess(args=['gdformat', '--check', '/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-...1/pytest-0/test_valid_unformatted_file_ch0/script.gd\n1 file would be reformatted, 0 files would be left unchanged.\n').stderr
FAILED tests/parser/test_executable.py::test_parsing_valid_file_succeeds - AssertionError: assert 2 == 0
 +  where 2 = len(['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'])
 +    where ['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'] = <built-in method splitlines of str object at 0x1108192c0>()
 +      where <built-in method splitlines of str object at 0x1108192c0> = '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.splitlines
 +        where '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = <built-in method decode of bytes object at 0x110819470>()
 +          where <built-in method decode of bytes object at 0x110819470> = b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.decode
 +            where b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = CompletedProcess(args=['gdparse', '/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test_parsi...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n').stderr
FAILED tests/parser/test_executable.py::test_parsing_valid_files_succeeds - AssertionError: assert 2 == 0
 +  where 2 = len(['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'])
 +    where ['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'] = <built-in method splitlines of str object at 0x110818f60>()
 +      where <built-in method splitlines of str object at 0x110818f60> = '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.splitlines
 +        where '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = <built-in method decode of bytes object at 0x110819110>()
 +          where <built-in method decode of bytes object at 0x110819110> = b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.decode
 +            where b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = CompletedProcess(args=['gdparse', '/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test_parsi...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n').stderr
FAILED tests/parser/test_executable.py::test_pretty_printing_valid_files_succeeds - AssertionError: assert 2 == 0
 +  where 2 = len(['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'])
 +    where ['/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.', '  import pkg_resources'] = <built-in method splitlines of str object at 0x110819980>()
 +      where <built-in method splitlines of str object at 0x110819980> = '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.splitlines
 +        where '/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:8...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = <built-in method decode of bytes object at 0x110819b30>()
 +          where <built-in method decode of bytes object at 0x110819b30> = b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n'.decode
 +            where b'/nix/store/js2vxmvsksvl4nwf5m7w0gzi32dkgr3b-gdtoolkit-4.3.3/lib/python3.13/site-packages/gdtoolkit/parser/parser.py:...for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n' = CompletedProcess(args=['gdparse', '-p', '/private/tmp/nix-build-gdtoolkit-4.3.3.drv-0/pytest-of-_nixbld1/pytest-0/test...or removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n  import pkg_resources\n').stderr
=========== 7 failed, 1047 passed, 335 skipped, 1 warning in 10.03s ============
```

### Additional context

_No response_

### System metadata

 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.6.0, macOS 14.6.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.28.4`
 - nixpkgs: `/nix/store/bgl6ldj5ihbwcq8p42z3a0qzgqafgk2b-source`

### Notify maintainers

@squarepear 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/gd/gdtoolkit_4/package.nix
+++ b/pkgs/by-name/gd/gdtoolkit_4/package.nix
@@ -27,14 +27,14 @@ let
 in
 python.pkgs.buildPythonApplication rec {
   pname = "gdtoolkit";
-  version = "4.3.3";
+  version = "4.3.4";
   format = "setuptools";
 
   src = fetchFromGitHub {
     owner = "Scony";
     repo = "godot-gdscript-toolkit";
     tag = version;
-    hash = "sha256-GS1bCDOKtdJkzgP3+CSWEUeHQ9lUcAHDT09QmPOOeVc=";
+    hash = "sha256-D67iwGGF3CrdAi/XKGVkusZlFCsMPIKdVpKDwcVQMrI=";
   };
 
   disabled = python.pythonOlder "3.7";
@@ -43,6 +43,7 @@ python.pkgs.buildPythonApplication rec {
     docopt
     lark
     pyyaml
+    radon
     setuptools
   ];
 
@@ -62,12 +63,13 @@ python.pkgs.buildPythonApplication rec {
   # The tests are not working on NixOS
   disabledTestPaths = [
     "tests/generated/test_expression_parsing.py"
-    "tests/gdradon/test_executable.py"
   ];
 
   pythonImportsCheck = [
     "gdtoolkit"
     "gdtoolkit.formatter"
+    "gdtoolkit.gd2py"
+    "gdtoolkit.gdradon"
     "gdtoolkit.linter"
     "gdtoolkit.parser"
   ];

